src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/RowConfigReader.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System.Diagnostics;
+
+namespace System.Net.NetworkInformation
+{
+    /// <summary> 
+    /// Helper for reading config files where each row is a key-value data pair.
+    /// The input key-values must not have any whitespace within them.
+    /// </summary>
+    internal struct RowConfigReader
```

Comment:
I may just move this into the Common directory next to the StringParser type. They're pretty similar, both are utility structs for reading config files.